tools/telemetry-generator/src/handlers/appInsightsExecutionTimeTestHandler.ts

```diff
@@ -14,49 +14,65 @@ module.exports = function handler(fileData, telemetryClient: TelemetryClient): v
 	for (const testData of fileData.benchmarks) {
 		const arithmeticMeanMetricName = `${fileData.suiteName}_${testData.benchmarkName}_arithmeticMean`;
 		try {
-			console.log(
-				`emitting metric ${arithmeticMeanMetricName} with value ${testData.customData["Period (ns/op)"]}`,
-			);
-			telemetryClient.trackMetric({
-				name: arithmeticMeanMetricName,
-				value: testData.customData["Period (ns/op)"],
-				namespace: "performance_benchmark_executionTime",
-				properties: {
-					buildId: process.env.BUILD_ID,
-					branchName: process.env.BRANCH_NAME,
-					category: "performance",
-					eventName: "Benchmark",
-					benchmarkType: "ExecutionTime",
-					suiteName: fileData.suiteName,
-					benchmarkName: testData.benchmarkName,
-					driverEndpointName: process.env.FLUID_ENDPOINTNAME ?? "",
-				},
-			});
+			const value = testData.customData["Period (ns/op)"];
+			if (Number.isNaN(Number.parseFloat(value))) {
+				console.error(
+					`skipping metric '${arithmeticMeanMetricName}' with value '${value}' as it is not a number`,
+				);
+			} else {
+				console.log(`emitting metric '${arithmeticMeanMetricName}' with value '${value}'`);
+				telemetryClient.trackMetric({
+					name: arithmeticMeanMetricName,
+					value,
+					namespace: "performance_benchmark_executionTime",
+					properties: {
+						buildId: process.env.BUILD_ID,
+						branchName: process.env.BRANCH_NAME,
+						category: "performance",
+						eventName: "Benchmark",
+						benchmarkType: "ExecutionTime",
+						suiteName: fileData.suiteName,
+						benchmarkName: testData.benchmarkName,
+						driverEndpointName: process.env.FLUID_ENDPOINTNAME ?? "",
+					},
+				});
+			}
 		} catch (error) {
-			console.error(`failed to emit metric ${arithmeticMeanMetricName}`, error);
+			console.error(`failed to emit metric '${arithmeticMeanMetricName}'`, error);
 		}
 
 		const marginOfErrorMetricName = `${fileData.suiteName}_${testData.benchmarkName}_marginOfError`;
 		try {
-			console.log(
-				`emitting metric ${arithmeticMeanMetricName} with value ${testData.customData["Margin of Error"]}`,
-			);
-			telemetryClient.trackMetric({
-				name: marginOfErrorMetricName,
-				value: testData.customData["Margin of Error"],
-				namespace: "performance_benchmark_executionTime",
-				properties: {
-					buildId: process.env.BUILD_ID,
-					branchName: process.env.BRANCH_NAME,
-					category: "performance",
-					eventName: "Benchmark",
-					benchmarkType: "ExecutionTime",
-					suiteName: fileData.suiteName,
-					benchmarkName: testData.benchmarkName,
-				},
-			});
+			const value = testData.customData["Margin of Error"];
+			if (value === undefined) {
+				console.error(`skipping metric '${marginOfErrorMetricName}' as it is undefined`);
```

Comment:
The check below for NaN already handles undefined correctly, so I'd say remove this extra one. That will also make the logic the same between this metric and the one above.

Reply:
Removed